MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult+Internal.swift

```diff
@@ -26,25 +26,84 @@ import Foundation
+                                                correlationId: correlationId ?? context.correlationId())) }
+            return
+        }
+
+        client.acquireTokenSilent(with: params) { [weak self] result, error in
+            guard let self = self else { return }
+
+            if let error = error as? NSError {
+                let accessTokenError = self.createRetrieveAccessTokenError(error: error, context: context)
+                Task { await delegate.onAccessTokenRetrieveError(error: accessTokenError) }
+                return
+            }
+
+            if let result = result {
+                let delegateDispatcher = CredentialsDelegateDispatcher(delegate: delegate, telemetryUpdate: nil)
+                let accessTokenResult = MSALNativeAuthTokenResult(accessToken: result.accessToken,
+                                                                  scopes: result.scopes,
+                                                                  expiresOn: result.expiresOn)
+                Task { await delegateDispatcher.dispatchAccessTokenRetrieveCompleted(result: accessTokenResult, correlationId: result.correlationId) }
+                return
+            }
+
+            Task { await delegate.onAccessTokenRetrieveError(error: RetrieveAccessTokenError(type: .generalError,
+                                                                                             correlationId: correlationId ?? context.correlationId())) }
         }
     }
 
-    func refreshData(authTokens: MSALNativeAuthTokens, account: MSALAccount) {
-        self.authTokens = authTokens
-        self.account = account
+    func createRetrieveAccessTokenError(error: NSError, context: MSALNativeAuthRequestContext) -> RetrieveAccessTokenError {
+        if let innerError = error.userInfo[NSUnderlyingErrorKey] as? NSError,
+           let message = innerError.userInfo[MSALErrorDescriptionKey] as? String,
+           let code = codeFromMSALError(error: innerError),
+           let correlationId = correlationIdFromMSALError(error: innerError) {
+            return RetrieveAccessTokenError(type: .generalError, message: message, correlationId: correlationId, errorCodes: [code])
+        }
+
+        if let message = error.userInfo[MSALErrorDescriptionKey] as? String,
+            let code = codeFromMSALError(error: error),
+            let correlationId = correlationIdFromMSALError(error: error) {
+            return RetrieveAccessTokenError(type: .generalError, message: message, correlationId: correlationId, errorCodes: [code])
+        }
+
+        return RetrieveAccessTokenError(type: .generalError,
+                                        message: error.localizedDescription,
+                                        correlationId: context.correlationId(),
+                                        errorCodes: [error.code])
+    }
+
+    private func codeFromMSALError(error: NSError) -> Int? {
+        return error.userInfo[MSALInternalErrorCodeKey] as? Int
```

Comment:
when we parse this error, we're not returning the API error code as we usually do, but the internal MSAL Objective-C code. For example, In the case of the API error AADSTS28000 ("Provided value for the input parameter scope is not valid because it contains more than one resource.") we return -42002 that it's an internal error. Or we parse the error code from the error message, or we just return an empty error code. So, we don't return a misleading information

Reply:
Removing error code from MSAL internal errors.